charts/ingress-nginx/templates/clusterrole.yaml

```diff
@@ -20,6 +20,7 @@ rules:
       - nodes
       - pods
       - secrets
+      - namespaces
```

Comment:
suggestion here: instead of allowing Ingress to watch any namespace (also I'm not a fan of allowing this for any secret, but this is another conversation) we can do something similar to:

https://github.com/kubernetes/ingress-nginx/blob/c4ad865ac437e20987b441834d884b2a7ec88442/charts/ingress-nginx/templates/clusterrole.yaml#L27-L33

But iteracting through the list of namespaces added in `--watch-namespace-selector`.

I'm not sure on how to do this, or maybe just allowing this permission as global (but without the resourceNames) IF the flag is setup.

Reply:
> But iterating through the list of namespaces added in --watch-namespace-selector.

It's possible to list all namespaces matched `--watch-namespace-selector` before `helm install`, but we cannot watch namespaces created with matched labels after installation. `--watch-namespace-selector` behaves just like `controller.scope.enabled=false`, instead of watching all namespaces, only a subset will be watched, we don't which namespaces yet. So cluster scope namespaces permission is a requirement here.